### PR TITLE
feat: Add PasswordInput to Labs components

### PR DESCRIPTION
**Pull request by drazik**
This is a new form of input field. But it needs to stabilized and maybe merged with current Input/Field. Labs seems to be a good place for it at the moment.

Related to https://github.com/cozy/cozy-ui/issues/1304

See https://github.com/cozy/cozy-settings/pull/269#discussion_r357225768

See https://drazik.github.io/cozy-ui/react/#!/PasswordInput

### CODE REVIEW

File: react/Labs/PasswordInput/Readme.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+This component also takes same props as the `Input` component.
```
Comment (ptbrowne) [inline react/Labs/PasswordInput/Readme.md]:
```suggestion
This component takes the same props as the `Input` component.
```

File: react/Labs/PasswordInput/helpers.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// Return given password srength as an object {percentage, label}
```
Comment (ptbrowne) [inline react/Labs/PasswordInput/helpers.js]: Please use a proper docstring.

s/Return/Return

File: react/Labs/PasswordInput/helpers.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  const charsets = [
+    // upper
+    { regexp: /[A-Z]/g, size: 26 },
+    // lower
+    { regexp: /[a-z]/g, size: 26 },
+    // digit
+    { regexp: /[0-9]/g, size: 10 },
+    // special
+    { regexp: /[!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]/g, size: 30 }
+  ]
```
Comment (ptbrowne) [inline react/Labs/PasswordInput/helpers.js]: Those could be extracted from the function. This way we do not create new regexps at each run.

File: react/Labs/PasswordInput/helpers.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  ) {
+    // between 33% and 66%
+    strengthPercentage = (passwordStrength * 66) / _at66percent
+    strengthLabel = 'moderate'
+  } else {
+    // passwordStrength > 192
+    strengthPercentage = (passwordStrength * 100) / _at100percent
+    if (strengthPercentage > 100) strengthPercentage = 100
+    strengthLabel = 'strong'
+  }
```
Comment (ptbrowne) [inline react/Labs/PasswordInput/helpers.js]: Seems like this logic is repeated and could be improved.

- strength labels could be put in a mapping linking label to password strength.

const strengthLabels = {
  weak: 0,
  moderate: 50,
  strong: 100
}

- levels could be extracted

- 100PercentStrength = strengthLabels.strong


I don't think there's the need to have _at33percent and _at66percent if the 100percent is known.

File: react/Labs/PasswordInput/helpers.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    return possibleChars
+  }, 0)
+
+  const passwordStrength =
+    Math.log(Math.pow(possibleChars, password.length)) / Math.log(2)
+
+  // levels
+  const _at33percent = 50
+  const _at66percent = 100
+  const _at100percent = 150
```
Comment (y-lohse) [inline react/Labs/PasswordInput/helpers.js]: I think I recognize this from zxcvbn, is it the same extracted from there? If so did you consider straight up using zxcvbn as a dependency?

File: react/Labs/PasswordInput/index.jsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+      value={strength.percentage}
+      className={cx(
+        styles.PasswordField__strength,
+        styles[`PasswordField__strength--${strength.label}`],
+        className
+      )}
+      {...rest}
+    />
+  )
+}
```
Comment (y-lohse) [inline react/Labs/PasswordInput/index.jsx]: Not sure how it looks, but would it be possible to use https://docs.cozy.io/cozy-ui/react/#percentageline by any chance?

File: react/Labs/PasswordInput/index.jsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+const PasswordInput = props => {
+  const { className, showStrength, error, ...rest } = props
+  const [hidden, setHidden] = useState(true)
+
+  return (
+    <div className={cx(styles.PasswordField, className)}>
```
Comment (y-lohse) [inline react/Labs/PasswordInput/index.jsx]: Nit, but we should avoid mixing `styles.PasswordField` and `styles['PasswordField']` 😁

File: react/Labs/PasswordInput/index.jsx (hunk 6 above)
Comment (y-lohse) [reply react/Labs/PasswordInput/index.jsx]: After looking at the css, I guess it's not similar. Might still be worth extracting it into a standalone component eventually.

File: react/Labs/PasswordInput/helpers.js (hunk 5 above)
Comment (drazik) [reply react/Labs/PasswordInput/helpers.js]: Actually I extracted it from [cozy-settings](https://github.com/cozy/cozy-settings/blob/master/src/lib/passwordHelper.js). I don't know if that's code extracted from zxcvbn, so I never considered using zxcvbn as a dependency.

File: react/Labs/PasswordInput/helpers.js (hunk 4 above)
Comment (drazik) [reply react/Labs/PasswordInput/helpers.js]: I added tests and created https://github.com/cozy/cozy-ui/issues/1315 so your suggestion can be done later with the confidence of the tests. I need to move forward quickly at the moment.

**Pull request merged**